### ISSUE TITLE
"PermissionError: [Errno 1] Operation not permitted" using `shutil.copy` on mounted filesystem for WSL

### ISSUE THREAD

**Issue by mguaypaq** [spinalcordtoolbox/spinalcordtoolbox#3900]
@mchen1110 is running into a file permission issue when trying to run `sct_run_batch` on Windows, at the step where it tries to make a copy of the script to run:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2b62a4ddfcd89b3fb18db90b3c926c1dd82c0ad5/spinalcordtoolbox/scripts/sct_run_batch.py#L482

This looks a lot like #3832, and probably has a very similar fix. I wanted to try doing a quick fix inside the existing SCT installation on the laptop, to see if I have the right fix (much easier to reproduce on the existing Windows system!), but we ran out of time today looking for where the SCT installation actually lives. We'll try again on a different day.

The copy-pasted terminal output is:
```
mchen@LAPTOP-FRUL1V8F:/mnt/c/Users/miche/Desktop/Neuropoly-Projects/basel-mp2rage-preprocessed$ sct_run_batch -jobs 1 -p
ath-data data_processed -path-output lesion-labeling -script correct_lesions.sh--
Spinal Cord Toolbox (git-master-b8d950994081b5a446e2b90c015e3bfdd3ca27ce)sct_run_batch -jobs 1 -path-data data_processed -path-output lesion-labeling -script correct_lesions.sh
--INFO SYSTEM
-----------
OS: linux (Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-debian-bullseye-sid)
CPU cores: Available: 8 | Threads used by ITK Programs: 1
RAM: Total 7872 MB | Available 6485 MB | Used 1035 MBINPUT ARGUMENTS
---------------
batch_log: sct_run_batch_log.txt
config: null
continue_on_error: 1
email_from: null
email_host: smtp.gmail.com:587
email_to: null
exclude: null
exclude_list: null
ignore_ses: false
include: null
include_list: null
itk_threads: 1
jobs: 1
path_data: data_processed
path_output: lesion-labeling
path_segmanual: .
script: correct_lesions.sh
script_args: ''
subject_prefix: sub-
v: 1
zip: falseSCRIPT
------
git commit: ?!?
git origin: ?!?
Copying script to output folder...
Traceback (most recent call last):
  File "/home/mchen/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_run_batch.py", line 598, in <module>
    main(sys.argv[1:])
  File "/home/mchen/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_run_batch.py", line 482, in main
    script_copy = os.path.abspath(shutil.copy(script, arguments.path_output))
  File "/home/mchen/spinalcordtoolbox/python/envs/venv_sct/lib/python3.7/shutil.py", line 249, in copy
    copymode(src, dst, follow_symlinks=follow_symlinks)
  File "/home/mchen/spinalcordtoolbox/python/envs/venv_sct/lib/python3.7/shutil.py", line 144, in copymode
    chmod_func(dst, stat.S_IMODE([st.st](http://st.st/)_mode))
PermissionError: [Errno 1] Operation not permitted: 'lesion-labeling/correct_lesions.sh'
```

**Comment by joshuacwnewton** [comment]
Small nitpick, but I think this is WSL, rather than native Windows?

> ```
> OS: linux (Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-debian-bullseye-sid)
> ```

As you say, this looks to be similar to #3832, in that it's a problem with:

* SCT being installed in the "WSL filesystem" (i.e. `/home/user/sct`)
* The data existing on the mounted "Windows filesystem" (i.e. `/mnt/c/Users/miche/Desktop/Neuropoly-Projects/basel-mp2rage-preprocessed`)

**Comment by joshuacwnewton** [comment]
As a side note, this might be a good opportunity to ask @mchen1110 to try out the [native Windows installation method](https://spinalcordtoolbox.com/user_section/installation/windows.html#native-windows-installation-recommended), rather than WSL? It'd be nice to have eyes on this new installation method internally, to ensure that it's fit for processing real-world data.

**Comment by mchen1110** [comment]
> As a side note, this might be a good opportunity to ask @mchen1110 to try out the [native Windows installation method](https://spinalcordtoolbox.com/user_section/installation/windows.html#native-windows-installation-recommended), rather than WSL? It'd be nice to have eyes on this new installation method internally, to ensure that it's fit for real-world usage.

I'm trying the Native Windows installation method and I encountered an error when I try to download the install_sct.bat script:

C:\Users\miche> python --version
Python 3.7.9

C:\Users\miche> git --version
git version 2.38.0.windows.1

### Downloading SCT source code (@ ) to \Users\miche\spinalcordtoolbox...
'git' is not recognized as an internal or external command,
operable program or batch file.
Failed with error #1.
Press any key to continue . . .

**Comment by joshuacwnewton** [comment]
It looks like there are two problems here:

1. `Downloading SCT source code (@ )`

This is an issue we encountered in https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3890. It was fixed on `master`, but the fix wasn't ported to the existing v5.7 installer. I'll do that right now!

2. `'git' is not recognized as an internal or external command, operable program or batch file.`

This is very odd to me, given that you were able to run `git --version` immediately prior to running the `install_sct.bat` script. I'll open a new issue for this!

**Comment by mchen1110** [comment]
Thank you!

**Comment by mguaypaq** [comment]
We ran into this problem again today.

While running a `manual_correction.py` script, we got two tracebacks involving `shutil.copy()` and "Operation not permitted". Similar to #3832, I tried replacing `shutil.copy()` with `shutil.copyfile()` in both places, only to be met with "IsADirectoryError", because the destination is a directory rather than a complete filename.

My next attempted fix was going to be to use [the slightly smarter logic from `shutil.copy()`](https://github.com/python/cpython/blob/0274a3bc3b1a479c766915a562db6fb19ef97daa/Lib/shutil.py#L418-L419) directly. But thankfully @sandrinebedard walked by and offered a better fix:

The problems are showing up because the output directory is a mounted shared drive from duke. But it's generally a bad idea to write results directly to duke, and the proper workflow is to simply use a local directory for the output. (The input can still be directly taken from duke, though.)
